script.py

```diff
@@ -0,0 +1,186 @@
+                                        .withColumn("NB_OF_STRUCT_PER_MOL", f.size(f.col("STRUCTURE_ID")))
+                                        .orderBy(f.desc("NB_OF_STRUCT_PER_MOL"))
+                                        )
+
+    unichem_molecule_struct_spark_df = (unichem_molecule_struct_spark_df
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL692')
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL1236970')
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL457299')
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL113178')
+                                        .select('MOLECULE_CHEMBL_ID', 'MOLECULE_PDB_ID', 'STRUCTURE_ID'))
+
+    # Save into a CSV
+    unichem_molecule_struct_spark_df.toPandas().to_csv("molecule_structure_target.csv", index=False, header=True)
+
+    # TIMER 2 STOP
+    stop_2 = timeit.default_timer()
+
+    # ----- GET TARGET ID -----
+    pdb_target_df = create_pdb_target_gene_df(args.ensembl, unichem_molecule_struct_spark_df)
+
+    # ----- STATISTICS -----
+    total_nb_unichem = unichem_df.count()
+    total_nb_molecule = molecule_df.count()
+    nb_mol_in_unichem = unichem_molecule_df.count()
+
+    total_nb_struct = unichem_molecule_struct_spark_df.withColumn('STRUCTURE_ID', f.explode('STRUCTURE_ID')).select(
+        'STRUCTURE_ID').distinct().count()
+
+    nb_target_pd = pdb_target_df.count()
+    nb_human_target = pdb_target_df.filter(pdb_target_df.GENE_ID.startswith('ENSG')).count()
+
+    count_distinct_target_pd_df = pdb_target_df.groupBy('GENE_ID').count().toPandas()
+    nb_molecule_without_struct = unichem_molecule_struct_spark_df.filter(f.size('STRUCTURE_ID') == 0).count()
+
+    nb_non_human_target = nb_target_pd - nb_human_target
+
+    # TIMER 1 STOP
+    stop_1 = timeit.default_timer()
+
+    statistics = [
+        ['Total molecules in UniChem database', total_nb_unichem],
+        ['Molecules in parquet', total_nb_molecule],
+        ['Molecules from parquet found in UniChem', nb_mol_in_unichem],
+        ['Percentage of parquet molecule found in UniChem', int(round((nb_mol_in_unichem / total_nb_molecule) * 100, 2))],
+        ['Total structures', total_nb_struct],
+        ['Molecules without structure', nb_molecule_without_struct],
+        ['Total targets', nb_target_pd],
+        ['Human target', nb_human_target],
+        ['None human target', nb_non_human_target],
+        ['Total time running', int(round(stop_1 - start_1, 2))],
+        ['Time for scrap structure on API', int(round(stop_2 - start_2, 2))]
+    ]
+
+    columns = ["Stats", "Count"]
+
+    stats_df = spark.createDataFrame(statistics, columns)
+
+    return stats_df, molecule_df, unichem_df, unichem_molecule_df, unichem_molecule_struct_spark_df, pdb_target_df
+
+
+if __name__ == '__main__':
+    main = main()
```

Comment:
Oh, this is really bad. Now you are overwriting the name `main` (the function) with `main` the variable. In general I would recommend not use the name `main` for variable given how consistently it is used to denote the main function.